### DISCUSSION TITLE
When not-found page is displayed, Navigating to an existing page continues to show not-found page.

### DISCUSSION THREAD

**Discussion by NeeshSamsi** [vercel/next.js#49516] [App Router]
I've never opened an issue here and didn't want to make a mistake, so I thought I could open a discussion instead.

Anyone else run into this issue where once the not-found page is shown, Navigating away to a different page using the Link component continues to show the not-found component.

![image](https://github.com/vercel/next.js/assets/72866286/ce4d6314-7a44-462a-b621-a258960852ce)
Here's the inital 404 response for /none as expected

![image](https://github.com/vercel/next.js/assets/72866286/4af6df18-2b3e-4220-9f0d-7d920f3fcd10)
Clicking on any Nav link or the links included in the 404 section going back to the / route still continues to display a 404
On refreshing the page at this stage, we get the expected behavior of displaying the / route
![image](https://github.com/vercel/next.js/assets/72866286/fd3d7c9f-aa92-4e77-9378-8bbce9f58f97)

Unsure if this is a common bug or maybe it's something I am doing wrong - Here's the GitHub repo - https://github.com/neeshsamsi/shan-blog

**Comment by icyJoseph** [comment]
Looks like a legit bug... Just managed to reproduce with `create-next-app`.

**Reply by icyJoseph** [reply]
Potential fix: https://github.com/vercel/next.js/issues/47862
Issue report: https://github.com/vercel/next.js/issues/49372

**Reply by NeeshSamsi** [reply]
Appreciate it. Thanks!
